dags/web_preaggregated.py

```diff
@@ -0,0 +1,459 @@
+    teams_sorted = sorted(pageview_volumes.items(), key=lambda item: item[1], reverse=True)
+
+    batches = []
+    current_batch = []
+    current_volume = 0
+
+    for team_id, volume in teams_sorted:
+        if current_volume + volume > target_batch_size and current_batch:
+            batches.append(current_batch)
+            current_batch = [team_id]
+            current_volume = volume
+        else:
+            current_batch.append(team_id)
+            current_volume += volume
+
+    if current_batch:
+        batches.append(current_batch)
+
+    return batches
+
+
+def _fetch_pageview_data(
+    context: dagster.AssetExecutionContext, cluster: dagster.ResourceParam[ClickhouseCluster]
+) -> dict:
+    """Fetch team pageview volume data from ClickHouse."""
+
+    def fetch_pageview_data(client: Client) -> dict:
+        return get_team_pageview_volumes(client)
+
+    pageview_volumes = cluster.any_host(fetch_pageview_data).result()
+    context.log.info(f"Retrieved pageview volume data for {len(pageview_volumes)} teams")
+    return pageview_volumes
+
+
+def _filter_and_batch_teams(
+    context: dagster.AssetExecutionContext, pageview_volumes: dict, team_ids: list[int]
+) -> tuple[dict, list[list[int]]]:
+    """Filter teams based on config if any is provided and create batches using the desired pageview volume per batch."""
+    if team_ids:
+        filtered_pageview_volumes = {team_id: pageview_volumes.get(team_id, 1) for team_id in team_ids}
+        context.log.info(f"Filtering to {len(filtered_pageview_volumes)} teams specified in config")
+    else:
+        filtered_pageview_volumes = pageview_volumes
+        context.log.info(f"Processing all {len(filtered_pageview_volumes)} teams with pageview data")
+
+    batches = split_teams_in_batches(filtered_pageview_volumes)
+    context.log.info(f"Created {len(batches)} batches based on pageview volume")
+
+    return filtered_pageview_volumes, batches
+
+
+def _process_single_team(
+    context: dagster.AssetExecutionContext,
+    cluster: dagster.ResourceParam[ClickhouseCluster],
+    team_id: int,
+    table_name: str,
+    sql_generator: Callable,
+    date_info: dict,
+    settings: str,
+) -> dict:
+    """Process a single team. This function is independent in the sense that it deletes previous existing data
+    so we can safely use to backfill periods and replace the current week data everyday so we can guarantee the data is always up to date.
+    """
+    start_time = datetime.now()
+
+    # Extract date information
+    partition_date = date_info["partition_date"]
+    start_date_str = date_info["start_date_str"]
+    end_date_str = date_info["end_date_str"]
+
+    # Delete existing data for this team and partition
+    delete_query = f"""
+    ALTER TABLE {table_name} DELETE WHERE
+    toDate(day_bucket) >= toDate('{partition_date}')
+    AND toDate(day_bucket) < toDate('{end_date_str}')
+    AND team_id = {team_id}
+    """
```

Comment:
I think we could probably use another table engine to handle the deduplication, but I assume that would be an eventual consistency approach, so I default to using a simple delete here so we can also run backfill jobs with more freedom while this is still under testing.